src/components/chip-set/chip-set.tsx

```diff
@@ -653,89 +575,82 @@ export class ChipSet {
         );
     };
 
-    private renderFilterChip(chip: Chip) {
-        return (
-            <div
-                class={`mdc-chip ${chip.selected ? SELECTED_CHIP_CLASS : ''}`}
-                role="row"
-                id={`${chip.id}`}
-            >
-                <span class="mdc-chip__checkmark">
-                    <svg class="mdc-chip__checkmark-svg" viewBox="-2 -3 30 30">
-                        <path
-                            class="mdc-chip__checkmark-path"
-                            fill="none"
-                            stroke="currentColor"
-                            d="M1.73,12.91 8.1,19.28 22.79,4.59"
-                        />
-                    </svg>
-                </span>
-                <span role="gridcell">
-                    <span
-                        role="checkbox"
-                        tabindex={this.disabled ? '-1' : '0'}
-                        aria-checked="false"
-                        class="mdc-chip__text"
-                    >
-                        {chip.text}
-                    </span>
-                </span>
-                {this.renderBadge(chip)}
-            </div>
-        );
-    }
-
     private renderInputChip(chip: Chip, index: number) {
+        const chipProps = this.getChipProps(chip, 'default');
+
         return [
-            <div
+            <limel-chip
+                key={chip.id}
                 class={{
-                    'mdc-chip': true,
-                    'mdc-chip--selected': this.inputChipIndexSelected === index,
-                    disabled: this.disabled,
+                    'can-be-removed': this.inputChipIndexSelected === index,
                 }}
-                role="row"
-                id={`${chip.id}`}
-                onClick={this.catchInputChipClicks}
-            >
-                {chip.icon ? this.renderChipIcon(chip) : null}
-                {this.renderChipLabel(chip)}
-                {this.renderChipRemoveButton(chip)}
-            </div>,
+                {...chipProps}
+                onRemove={this.removeChip(chip.id)}
+            />,
             this.renderDelimiter(),
         ];
     }
 
-    private catchInputChipClicks(event) {
-        event.stopPropagation();
+    private removeChip = (id: string | number) => () => {
```

Comment:
I'm a little worried that this construct might cause _very_ many functions to be created, since it will create a new function not just for each chip, but each time a chip is re-rendered. I don't really know how often that might be. Perhaps it's only a few times in a chip's lifetime, but if it happens for all the chips anytime anything in the chip-set is updated, that could be a little bit nasty…

We'd avoid this completely if the chip could accept some kind of identifier, and then emit that same value on the `remove` event. It doesn't mean we have to make that property required. It could still be optional, and if it's set, we attach it to the `remove` event.

(Ping me if you would like me to help implement it.)

Reply:
Why didn't we pass the chip object to `<limel-chip>` instead of passing many properties? I could not understand the reason behind.

Reply:
yes, I also don't really understand why limel-chip is importing the `Chip` interface, but also repeats everything from the Chip interface, as a prop inside itself. But generally I don't have a good understanding of theses things. I just assumed that was needed. Maybe because limel-chip might become a public component in future? You guys discussed this here: https://github.com/Lundalogik/lime-elements/pull/2707#discussion_r1452633337

Reply:
This runs only once when you click remove button otherwise I don't think it runs each time a chip is re-rendered. 

>We'd avoid this completely if the chip could accept some kind of identifier, and then emit that same value on the remove event. It doesn't mean we have to make that property required. It could still be optional, and if it's set, we attach it to the remove event.

That would be better to have an identifier prop. I did not understand why it was not passed from the beginning. That is why I used a curry here. You decide the best one I will add a fixup if you are not super happy

Reply:
> This runs only once when you click remove button otherwise I don't think it runs each time a chip is re-rendered.

The inner function will run only when the button is clicked. The _outer_ function, that creates and returns the inner function, will run each time the template is rendered.

I'll experiment a little with it.

Reply:
I added this `console.log` to the outer function:

<img width="507" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/154725/0112c6c9-9b64-4e27-82d2-ad8a21e504c6">

…and it's called a _lot…_

https://github.com/Lundalogik/lime-elements/assets/154725/75fc7d14-df24-4b6a-9cc7-14bdba43cdfa

Reply:
Nice argument. Should we have an identifier prop in the child component then?

Reply:
Yes, I think that's the way to go 👍

But it should be optional, so people who might use the chip standalone (once we make it public), doesn't have to supply one.

Reply:
https://github.com/Lundalogik/lime-elements/pull/2777/commits/33e8bccc42a531327a5246f3f046b70682f71be0 I leave it for you if you don't mind. Both @Kiarokh and @adrianschmidt you guys can update the code if you find it unreadable or difficult to understand.

Reply:
@adrianschmidt are you aware that you removed this fixup when you squash it?

Reply:
Oh, no, I'm sorry! I don't know how that happened 🙈

Reply:
I didn't see that you had added that, so I was starting to fix it myself…

I'll add it back in 👍

Reply:
I had made some additional changes, so I combined your changes and mine 👍

I've pushed them now.

Reply:
Hmm… deleting chips with the keyboard doesn't work at the moment. I must have missed something.

Reply:
Ah, now I see why you had added a `handleRemoveChip` 😄

Reply:
⚡